### DISCUSSION TITLE
Testing custom transformers

### DISCUSSION THREAD

**Discussion by rpungello** [spatie/laravel-data#745] [Q&A]
I have an application that relies on the PHP decimal extension for accurate decimal arithmetic, and have created a transformer to allow `Decimal` objects to be used as data properties.
I was looking to create a test for this transformer that verifies it is working correctly. I noticed the built-in transformers have tests, but they rely on a class (`Spatie\LaravelData\Tests\Factories\FakeDataStructureFactory`) that is seemingly not present on the release version of the package.

Is there a recommended way for package-specific transformers to be tested? Mainly how to create the `DataProperty` parameter required by the `transform()` function.

**Comment by rubenvanassche** [comment]
You can always do app(DataConfig::class)->getDataClass(YOUR DATA CLASS HERE) and then find the correct property within the structure. Or just copy this into your project: https://github.com/spatie/laravel-data/blob/174b6e5dc7e32fba59819b7f33a02482f7cff1f1/tests/Factories/FakeDataStructureFactory.php#L21
